File: outliner/tests/fixtures/py/sample.py
#!/usr/bin/env python3
"""Module docstring."""

__version__ = "1.0.0"
__all__ = [
    "Animal",
    "make_animal",
]

MAX_SIZE: int = 100


# Represents any animal
class Animal:
    """Base animal class."""

    kind = "unknown"

    def __init__(self, name: str, kind: str = "unknown") -> None:
        self.name = name
        self.kind = kind

    def speak(self) -> str:
        return f"{self.name} says hello"

    @property
    def display_name(self) -> str:
        return self.name.title()

    @staticmethod
    def species() -> str:
        return "Animalia"

    @classmethod
    def from_dict(cls, data: dict) -> "Animal":
        return cls(data["name"], data.get("kind", "unknown"))


class Dog(Animal):
    """A dog."""

    def speak(self) -> str:
        return f"{self.name} says woof"

    def fetch(
        self,
        item: str,
        distance: float = 1.0,
    ) -> bool:
        return True


def make_animal(
    name: str,
    kind: str = "unknown",
) -> "Animal":
    """Create an animal."""
    return Animal(name, kind)


async def fetch_animal(url: str) -> dict:
    """Fetch animal data."""
    pass


def complex_factory(
    name: str,
    size: int = 0,
    verbose: bool = False,
) -> "Animal":
    return Animal(name)


def _helper(x: int) -> int:
    return x * 2

File: outliner/tests/fixtures/rb/sample.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'json'
require_relative './utils'

# A module for animals
module Animals
  VERSION = '1.0'

  # Represents any animal
  class Animal
    attr_accessor :name, :kind
    attr_reader :id

    # @param name [String]
    def initialize(name, kind: 'unknown')
      @name = name
      @kind = kind
      @id = rand(1000)
    end

    # Speak in animal's voice
    def speak
      "#{@name} says hello"
    end

    def display_name
      @name.upcase
    end

    def self.create(name)
      new(name)
    end

    private

    def secret_method
      'secret'
    end
  end

  # A dog
  class Dog < Animal
    def initialize(name)
      super(name, kind: 'dog')
    end

    def speak
      "#{@name} says woof"
    end

    def fetch(
      item,
      distance: 1.0
    )
      item.to_s.length > distance
    end
  end

  def self.make_animal(name, kind = 'unknown')
    Animal.new(name, kind)
  end
end

# Standalone helper
def standalone_helper(x)
  x * 2
end
